### PR TITLE
D2IQ-66357: repositions Dropdownable on scroll

### PR DESCRIPTION
**Pull request by mperrotti**
Closes D2IQ-66357

## Testing
- Components that use Dropdownable should update their position on scroll. The Dropdownable overlay should be hidden when the trigger is scrolled out of view.

## Trade-offs
PopperJS has been added as a dependency. It's pretty small, but it has a lot of very useful features.

## Screenshots
RepositionDropdownableOnScroll

## Checklist for PR creator

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [x] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [x] Info for applicable sections above is provided

### CODE REVIEW

File: packages/dropdownable/components/Dropdownable.tsx [hunk 1]
```diff
@@ -86,17 +88,26 @@ class Dropdownable extends React.Component<DropdownableProps, State> {
 
   componentDidMount() {
     resizeEventManager.add(this.setPositionFromCurrentProps);
+    window.addEventListener("scroll", this.hideDropdownable, true);
+    // windowEventManager.add("scroll", this.hideDropdownable, true);
```
Comment (GeorgiSTodorov) [inline packages/dropdownable/components/Dropdownable.tsx:92]: I think this was forgotten. I am removing it.

**Comment by pierrebeitz** [comment]
i don't feel too good about this change. 

i think i have not seen this behaviour in any other library for popovers yet (at least for popovers that i opened with a click).
the other thing is that we now have a "partially controlled component" - which is not the most straightforward thing to work with from my perspective.

can we maybe solve the original issue differently? i think issues like there are the reason why e.g. bootstrap ships with [popper.js](https://popper.js.org/).

**Comment by mperrotti** [comment]
@pierrebeitz, I think you're right about the "partially controlled component" being undesirable. I'll look into alternative ways to solve this problem. I'm very open to any recommendations.

I've seen this "close the dropdown on scroll" behavior before, and I'd prefer we keep it. It helps us avoid accessing DOM node measurements, which causes a repaint.

**Comment by mperrotti** [comment]
I haven't come up with anything I'm happy with for hiding on scroll. 😅 
I'm considering using Popper.js (probably via react-popper) to power Dropdownable and keep it performant.

**Comment by mperrotti** [comment]
@pierrebeitz - it's referring to the "Expand/Collapse" button on the left

**Pull request merged**